### ISSUE TITLE
Quick registry creates duplicate outputs

### ISSUE THREAD

**Issue by pb-cdunn** [PacificBiosciences/pbcommand#16]
The dummy target fixed the problem in PacificBiosciences/pbsmrtpipe#21 , but it created a new one.

``` python
@dev_register("pipe_falcon", "Falcon Pipeline", tags=("local", ))
def get_task_falcon_local_pipeline():
    b1 = [
          ('falcon_ns.tasks.task_falcon_config:0',        'falcon_ns.tasks.task_falcon_build_rdb:0'),
          ('falcon_ns.tasks.task_falcon_make_fofn_abs:0', 'falcon_ns.tasks.task_falcon_build_rdb:1'),
         ]
```

```
output_files: [
  '..../testkit-data/testkit_falcon/job_output/tasks/falcon_ns.tasks.task_falcon_build_rdb-0/file.txt',
  '..../testkit-data/testkit_falcon/job_output/tasks/falcon_ns.tasks.task_falcon_build_rdb-0/file.txt']
```

In other words, the output is just the same file twice!

For inputs, this is the correct behavior, outputs need to be uniquified. Better would be to accept `OutputFileType` in addition to `FileType`. In `pbcommand/cli/quick.py`:

``` python
 36 def _file_type_to_output_file_type(file_type):
 37     return OutputFileType(file_type.file_type_id,
 38                           "Label " + file_type.file_type_id,
 39                           repr(file_type),
 40                           "description for {f}".format(f=file_type),
 41                           file_type.default_name)
...
    class Registry:
 79     def __call__(self, tool_id, version, input_types, output_types, options=None, nproc=1, is_distributed=True):
 80         def _w(func):
...
 95             output_file_types = [_file_type_to_output_file_type(ft) for ft in otypes]

```

A Factory of OutputFileType would be even more convenient.

I could also just update the JSON ToolContract, but I am unsure what other changes would be required. Also, I don't see where to specify the filename, and I don't know when "default_name" is not used.

So for now, I'll copy and modify the code above.

**Comment by pb-cdunn** [comment]
> And can you post the repo with your TCs and pipelines. I need both of those to understand whats going on.

See [reference above](https://github.com/pb-cdunn/pbsmrtpipe/commit/70b526e7d503bf14006917b510af4c1ee6455a04).

I have something working now, but I'm happy to switch to something simpler when available.

**Comment by pb-cdunn** [comment]
Interesting. It runs, but it still doesn't do the right thing. I now have this in my generated TC, `pbsmrtpipe/registered_tool_contracts/falcon_ns.tasks.task_falcon_build_rdb_tool_contract.json`:

``` json
        "output_types": [
            {
                "default_name": "script.sh",
                "description": "description for <FileType id=PacBio.FileTypes.txt name=file.txt >",
                "file_type_id": "PacBio.FileTypes.txt",
                "id": "Label PacBio.FileTypes.txt",
                "title": "<FileType id=PacBio.FileTypes.txt name=file.txt >"
            },
            {
                "default_name": "DUMMY1.txt",
                "description": "description for <FileType id=PacBio.FileTypes.txt name=file.txt >",
                "file_type_id": "PacBio.FileTypes.txt",
                "id": "Label PacBio.FileTypes.txt",
                "title": "<FileType id=PacBio.FileTypes.txt name=file.txt >"
            }
        ],
```

But in the stdout for this task, I see this:

```
  output_files: [
'..../testkit-data/testkit_falcon/job_output/tasks/falcon_ns.tasks.task_falcon_build_rdb-0/file.txt',
'..../testkit-data/testkit_falcon/job_output/tasks/falcon_ns.tasks.task_falcon_build_rdb-0/file.txt']
```

Weird! I don't see where it even gets those names.

**Comment by mpkocher** [comment]
It's the resolver in pbcommand. Can you take a look at suggest a fix? I've added a new end-to-end test that should help (dev_txt_hello).

The new TC model (and pbcommand package) is relatively news so there are features in not converged between the two projects, such as the resolving mechanism.

**Comment by mpkocher** [comment]
FIxed in 0.2.12
